tests/framework/utils_vhost_user_backend.py

```diff
@@ -33,3 +50,12 @@ def spawn_vhost_user_backend(vm, host_mem_path, socket_path, readonly=False):
         os.chown(socket_path, uid, gid)
 
     return proc
+
+
+def resize(new_size, backend="crosvm"):
+    """Resize vhost-user-blk drive and send config change notification"""
+
+    if backend != "crosvm":
+        assert False, f"backend `{backend}` does not support resizing"
+
+    utils.run_cmd(f"crosvm disk resize 0 {new_size} {CROSVM_CTR_SOCKET}")
```

Comment:
Would it make sense to have a few classes to keep it organized?

```python
class VhostUserBackend(ABC):
    def __init__(self, socket: Path):
        self.socket = socket

    @abstractmethod
    def spawn(self):
        pass

    def resize(self):
        raise NotImplementedError()

    @staticmethod
    def with_backend(backend, *args, **kwargs):
        if backend == "qemu":
            return VhostUserBackendQEmu(*args, **kwargs)
        ...

class VhostUserBackendQEmu(VhostUserBackend):
    def spawn(self):
        pass

class VhostUserBackendCrosvm(VhostUserBackend):
    def resize(self):
        pass

> VhostUserBackend.with_backend("qemu", socket="/my.sock")
> <__main__.VhostUserBackendQEmu object at 0x7f70773300d0>
```

Reply:
do you think that would be worth it, provided that we only have 2 methods (spawn and resize), and resize will only be supported by one of the backends?